### PR TITLE
Feature/Forget Password API

### PR DESCRIPTION
**Pull request by SyedSamrozeAli**

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -16,3 +17,8 @@
 Route::get('/', function () {
     return view('welcome');
 });
+
+Route::get('/users/forget-password', [LoginRegisterControllers::class, 'forgetPassword']);
```
Comment (ZeeshanHabib7) [inline routes/web.php]: this should be in api routes

File: routes/web.php [hunk 2]
```diff
@@ -16,3 +17,8 @@
...
+Route::post('/users/send-email', [LoginRegisterControllers::class, 'sendEmail'])->name('sendEmail');
```
Comment (ZeeshanHabib7) [inline routes/web.php]: this also should be in api routes

File: app/Http/Controllers/LoginRegisterControllers.php [hunk 3]
```diff
@@ -54,4 +59,65 @@ protected function getToken(Request $request)
...
+                $domain = URL::to('/');
+                $url = $domain . '/users/reset-password/' . $user->id;
+                $data = [
+                    "url" => $url,
+                    "email" => $user->email,
+                    "title" => "Reset Password Link",
+                    "body" => "Please click on the below button to reset your password",
+                ];
+
+                Mail::to('k224187@nu.edu.pk')->send(new ForgetPasswordMail($data));
```
Comment (ZeeshanHabib7) [inline app/Http/Controllers/LoginRegisterControllers.php]: why there is a hard coded email here?

File: app/Http/Controllers/LoginRegisterControllers.php [hunk 4]
```diff
@@ -54,4 +59,65 @@ protected function getToken(Request $request)
...
+        } catch (\Exception $e) {
+            return back()->withErrors(['error' => $e->getMessage()]);
+        }
+    }
+
+    public function resetPasswordPageLoad($id)
+    {
+        try {
+
+            $user = User::where('id', $id)->first();
```
Comment (ZeeshanHabib7) [inline app/Http/Controllers/LoginRegisterControllers.php]: use firstOrfail here or you can also use findOrfail() too

File: app/Http/Controllers/LoginRegisterControllers.php [hunk 5]
```diff
@@ -54,4 +59,65 @@ protected function getToken(Request $request)
...
+            return errorResponse($e->getMessage());
+        }
+
+    }
+
+    public function resetPassword(ResetPasswordRequest $request)
+    {
+        try {
+
+            $user = User::where('email', $request->email)->first();
```
Comment (ZeeshanHabib7) [inline app/Http/Controllers/LoginRegisterControllers.php]: use firstOrfail here too

File: resources/views/forget-password.blade.php [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        @endif
+        @if ($errors->any())
+            <p style="font-size: 1vw; color:red; margin-top:30px;">{{ $errors->first() }}</p>
+        @endif
+        <p id="timerr"></p>
+    </div>
+
+    <script>
+
+        document.addEventListener('DOMContentLoaded', function () {
```
Comment (ZeeshanHabib7) [inline resources/views/forget-password.blade.php]: move the script file to a separate file please

File: app/Http/Controllers/LoginRegisterControllers.php (hunk 3 above)
Comment (SyedSamrozeAli) [reply app/Http/Controllers/LoginRegisterControllers.php]: Sir all the emails in DB were fake so for checking purpose I hardcoded this and forget to remove.

Comment (ZeeshanHabib7) [reply app/Http/Controllers/LoginRegisterControllers.php]: ok but please take care next time

**Pull request merged**